### PR TITLE
[CI:DOCS] buildah image should not enable fuse-overlayfs for rootful mode

### PR DESCRIPTION
**Pull request by rhatdan**
#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by rhatdan** [comment]
@giuseppe @TomSweeneyRedHat @vrothberg @flouthoc PTAL

**Review by giuseppe — APPROVED** [review]
LGTM

**Review by flouthoc — APPROVED** [review]
LGTM
/lgtm
/approve

**Comment by TomSweeneyRedHat** [comment]
Looks like a rebase is needed @rhatdan

File: contrib/buildahimage/Containerfile [hunk 1]
```diff
@@ -67,8 +67,7 @@ ADD ./containers.conf /etc/containers/
 
 # Copy & modify the defaults to provide reference if runtime changes needed.
 # Changes here are required for running with fuse-overlay storage inside container.
```
Comment (TomSweeneyRedHat) [inline contrib/buildahimage/Containerfile:69]: I think, but dont' know for sure, that this line still holds.  But wanted to double check.

**Comment by rhatdan** [comment]
Ignoring the rpm-build issues, since they are not related to this PR.

**Pull request merged**